### ISSUE TITLE
raidboss: new timeline keyword `hold`

### ISSUE THREAD

**Issue by quisquous** [quisquous/cactbot#5143]
# Motivation

One thing that is implicitly done in timelines today that I think would be better done explicitly is that when there's a branch/phase/loop, timeline authors add "enough" time after that before starting a new section.  For example, https://github.com/quisquous/cactbot/blob/main/ui/raidboss/data/06-ew/dungeon/the_sildihn_subterrane.txt#L164-L168

There's a few issues here with what is "enough time":

1) If the sync is missed for whatever reason, then eventually the timeline will run into the section afterwards, and things will look very wrong (as opposed to just showing nothing).

2) `ShowTimerBarAtSeconds` is also configurable, so somebody could put this extremely high.  In general, timeline authors assume "thirty seconds" of lookahead, but this has always just been convention.  If somebody puts the number very high, it could very easily show abilities in the timeline that shouldn't be there.  (Also if every timeline used `hold` we could just make this infinity and not have an option.)

(Also, I am not tied to `hold` as the word, and would take other options.  Other options seem like `stop` or even `break`.)

I think this isn't the hugest of issues, but I think it would make timeline intentions more clear.

# Proposal

Add a new keyword that goes on a line on its own, e.g.
```txt
1362.0 hold
```

Once the timeline gets there, time no longer passes.  In addition, no abilities past that line are shown in the future.  If there is sync with a window that overlaps the hold time, jumping to a new time will start time passing again.  The idea would be that you'd put this at the end of sections or after multi-phase jumps to indicate that time shouldn't pass normally over that `hold`.

I could also just see a proposal for `hold` on a line by its own with no time (inheriting the time of the line before it), since it doesn't really make sense to have a time on its own.

cc @JLGarber @valarnin @panicstevenson for timeline opinions

**Comment by JLGarber** [comment]
Makes sense to me. There aren't necessarily a lot of places where this would be useful, but it would definitely be nice to just standardize everything to Do It Right. This is particularly important in light of `ShowTimerBarAtSeconds` existing.

As far as syntax concerns, I think I prefer having the keyword holding an explicit timestamp. One thing the timeline syntax currently allows for is arbitrary ordering of lines, and that would break if this were implemented in a relative fashion. I'm not sure anyone actually *uses* that, but I'm not sure there's a good reason to break it either.
